{"abstract":[{"text":"Common errors thrown by ","type":"text"},{"code":"ByteToMessageDecoder","type":"codeVoice"},{"text":"s.","type":"text"}],"relationshipsSections":[{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/s5ErrorP","doc:\/\/co.pointfree.VaporRouting\/s8SendableP"],"kind":"relationships","type":"conformsTo","title":"Conforms To"}],"schemaVersion":{"major":0,"patch":0,"minor":3},"metadata":{"roleHeading":"Enumeration","navigatorTitle":[{"kind":"identifier","text":"ByteToMessageDecoderError"}],"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"ByteToMessageDecoderError"}],"modules":[{"name":"NIOCore"}],"externalID":"s:7NIOCore25ByteToMessageDecoderErrorO","title":"ByteToMessageDecoderError","symbolKind":"enum","role":"symbol"},"topicSections":[{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/PayloadTooLargeError"],"title":"Structures"},{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/dataReceivedInErrorState(_:_:)","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/leftoverDataWhenDone(_:)"],"title":"Enumeration Cases"}],"identifier":{"url":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError","interfaceLanguage":"swift"},"sections":[],"primaryContentSections":[{"kind":"declarations","declarations":[{"languages":["swift"],"platforms":["Linux"],"tokens":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"ByteToMessageDecoderError"}]}]}],"variants":[{"paths":["\/documentation\/niocore\/bytetomessagedecodererror"],"traits":[{"interfaceLanguage":"swift"}]}],"hierarchy":{"paths":[["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore"]]},"kind":"symbol","references":{"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError/leftoverDataWhenDone(_:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"leftoverDataWhenDone","kind":"identifier"},{"text":"(","kind":"text"},{"text":"ByteBuffer","kind":"typeIdentifier","preciseIdentifier":"s:7NIOCore10ByteBufferV"},{"text":")","kind":"text"}],"abstract":[{"text":"This error can be thrown by ","type":"text"},{"code":"ByteToMessageDecoder","type":"codeVoice"},{"text":"s if there was unexpectedly some left-over data when the","type":"text"},{"text":" ","type":"text"},{"code":"ByteToMessageDecoder","type":"codeVoice"},{"text":" was removed from the pipeline or the ","type":"text"},{"code":"Channel","type":"codeVoice"},{"text":" was closed.","type":"text"}],"type":"topic","url":"\/documentation\/niocore\/bytetomessagedecodererror\/leftoverdatawhendone(_:)","title":"ByteToMessageDecoderError.leftoverDataWhenDone(_:)","role":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/leftoverDataWhenDone(_:)","kind":"symbol"},"doc://co.pointfree.VaporRouting/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/co.pointfree.VaporRouting\/s8SendableP"},"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError":{"abstract":[{"text":"Common errors thrown by ","type":"text"},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"text":"s.","type":"text"}],"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError","role":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"ByteToMessageDecoderError","kind":"identifier"}],"title":"ByteToMessageDecoderError","type":"topic","navigatorTitle":[{"text":"ByteToMessageDecoderError","kind":"identifier"}],"kind":"symbol","url":"\/documentation\/niocore\/bytetomessagedecodererror"},"doc://co.pointfree.VaporRouting/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/co.pointfree.VaporRouting\/s5ErrorP"},"doc://co.pointfree.VaporRouting/documentation/NIOCore":{"type":"topic","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore","kind":"symbol","title":"NIOCore","abstract":[],"url":"\/documentation\/niocore","role":"collection"},"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError/dataReceivedInErrorState(_:_:)":{"url":"\/documentation\/niocore\/bytetomessagedecodererror\/datareceivedinerrorstate(_:_:)","type":"topic","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/dataReceivedInErrorState(_:_:)","role":"symbol","title":"ByteToMessageDecoderError.dataReceivedInErrorState(_:_:)","kind":"symbol","abstract":[{"type":"text","text":"More data has been received by a "},{"type":"codeVoice","code":"ByteToMessageHandler"},{"type":"text","text":" despite the fact that an error has previously been"},{"type":"text","text":" "},{"type":"text","text":"emitted. The associated "},{"type":"codeVoice","code":"Error"},{"text":" is the error previously emitted and the ","type":"text"},{"type":"codeVoice","code":"ByteBuffer"},{"text":" is the extra data that has","type":"text"},{"text":" ","type":"text"},{"text":"been received. The common cause for this error to be emitted is the user not having torn down the ","type":"text"},{"type":"codeVoice","code":"Channel"},{"text":" ","type":"text"},{"text":"after previously an ","type":"text"},{"type":"codeVoice","code":"Error"},{"text":" has been sent through the pipeline using ","type":"text"},{"type":"codeVoice","code":"fireErrorCaught"},{"text":".","type":"text"}],"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"dataReceivedInErrorState","kind":"identifier"},{"text":"(","kind":"text"},{"text":"Error","preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"ByteBuffer","preciseIdentifier":"s:7NIOCore10ByteBufferV","kind":"typeIdentifier"},{"text":")","kind":"text"}]},"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError/PayloadTooLargeError":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/PayloadTooLargeError","url":"\/documentation\/niocore\/bytetomessagedecodererror\/payloadtoolargeerror","type":"topic","title":"ByteToMessageDecoderError.PayloadTooLargeError","abstract":[{"type":"text","text":"This error can be thrown by "},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":"s if the incoming payload is larger than the max specified."}],"role":"symbol","navigatorTitle":[{"kind":"identifier","text":"PayloadTooLargeError"}],"kind":"symbol","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"PayloadTooLargeError"}]}}}